Add note to astro-cli docker setup (#3191)
astronomer/docs: astro/cli/install-cli.md

By default, the Astro CLI uses Docker as its container management engine for running Airflow locally. However, if your organization uses Podman to run and manage containers, you can configure the Astro CLI to use it instead. See [Run the Astro CLI with Podman](cli/configure-cli.md?tab=mac#run-the-astro-cli-using-podman) for prerequisites and configuration steps.

:::info

> When using Docker Desktop make sure that the setting "Use containerd for pulling and storing images" is turned off. Otherwise, you might run into issues when executing the `astro deploy` command, such as:

Reviewer: ```suggestion
When you use Docker Desktop, ensure that the [**Use containerd for pulling and storing images**](https://docs.docker.com/desktop/containerd/#turn-on-the-containerd-image-store-feature) setting is turned off. Otherwise, you might receive errors when you run `astro deploy` such as:

```

> ```

Reviewer: ```suggestion
```text
```

> push access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed
> ```
> or:
> ```
> Unable to find image 'barren-ionization-0185/airflow:latest' locally
> Error response from daemon: pull access denied for barren-ionization-0185/airflow, repository does not exist or may require 'docker login'

Reviewer: ```suggestion
push access denied, repository does not exist or may require authorization: server message: insufficient_scope: authorization failed
...
Unable to find image 'barren-ionization-0185/airflow:latest' locally
Error response from daemon: pull access denied for barren-ionization-0185/airflow, repository does not exist or may require 'docker login'
```

Author: The first error was when running with `-f` i.e. `astro deploy <deployment-id> -f`

The second error was just `astro deploy <deployment-id>`
